jupyterhub/schema.yaml

```diff
@@ -212,6 +212,89 @@ properties:
           the `--values` or `-f` flag. During merging, lists are replaced while
           dictionaries are updated.
           ```
+      extraFiles: &extraFiles
+        type: string
+        description: |
+          A dictionary with extra files that should be injected into the pod's
+          container on startup.
+
+          ```yaml
+          hub:
+            extraFiles:
+              <implicit file name>:
+                # name is an optional way to override the implicit file name
+                # that must be unique.
+                name: <explicit file name>
+
+                # mode is by default 0644 and you can optionally override it
+                # either by octal notation (example: 0400) or decimal notation
+                # (example: 256).
+                mode: <file system permissions>
+
+                # mountDir is required and must be the directory you want to
+                # mount the file.
+                mountDir: <directory path>
```

Comment:
Since `mountDir` is mandatory have you thought about also requiring the name in the same property (`mountPath`) and always ignoring the key, so there's a bit less logic to deal with?

Reply:
Good point! I removed the explicit `name` configuration entirely, removed `mountDir` and added `mountPath` as a required config that should include the file name as well.